libraries/botbuilder-dialogs-adaptive/src/recognizers/recognizerSet.ts

```diff
@@ -63,13 +64,15 @@ export class RecognizerSet extends Recognizer implements RecognizerSetConfigurat
 
                 // merge intents
                 for (const intentName in result.intents) {
-                    const intentScore = result.intents[intentName].score;
+                    const intent = result.intents[intentName];
+                    const intentScore = intent.score;
                     if (recognizerResult.intents.hasOwnProperty(intentName)) {
                         if (intentScore < recognizerResult.intents[intentName].score) {
                             continue;
                         }
                     }
-                    recognizerResult.intents[intentName] = { score: intentScore };
+                    // recognizerResult.intents[intentName] = { score: intentScore };
+                    recognizerResult.intents[intentName] = JSON.parse(JSON.stringify(intent));;
```

Comment:
Is this deep clone style necessary?

Reply:
Yeah, as intents come back as `Record<string, IntentScore>`, and `IntentScore` can have a score + whatever other property.

roughly
```
`RecognizerResult {
    `intents`: {
        string: IntentScore {
           score: number
           string: unknown
     }
}
```

So when we created, for example, a `RecognizerSet` with a `RegexRecognizer`, it could include an intent pattern like:
```ts
new IntentPattern('codeIntent', '(?<code>[a-z][0-9])')
```
And without the change we were losing the pattern in the `IntentScore` results that we got in C# results

![image](https://user-images.githubusercontent.com/35248895/107696435-cd31f780-6c66-11eb-9d7c-0b234ccb2df4.png)

Reply:
That makes sense - there's a difference, though, between doing `const foo = JSON.parse(JSON.stringify(someObj))` and `const foo = { ...someObj }` (or the `Object.assign` variant). My question is - is it necessary to parse/stringify bit?

Reply:
Ah, I understand now. And going through refactoring to for-of loops as requested, tried doing away with the deep cloning, and things work just fine without it 👍